src/app/google-api.service.ts

```diff
@@ -147,4 +142,21 @@ public async getCookie(): Promise<boolean>{
   }
   return false;
 }
+
+
+public createCookie(){
+  const token = gapi.client.getToken();
+  const tokenString = JSON.stringify(token);
+  const currentTime = new Date();
+  const expiryTime = currentTime.getMilliseconds() + (60*59*1000);
```

Comment:
instead of using your own expiry time, can you use the expires_in with the token?

just in case google decides to issue you with a token that expires earlier for some reason